core/sdk-core/src/main/java/software/amazon/awssdk/core/endpointdiscovery/EndpointDiscoveryRefreshCache.java

```diff
@@ -109,4 +114,30 @@ public CompletableFuture<EndpointDiscoveryEndpoint> discoverEndpoint(EndpointDis
     public void evict(String key) {
         cache.remove(key);
     }
+
+    String getKey(String accessKey, EndpointDiscoveryRequest request) {
```

Comment:
Should it be private? It's a bit code smell to expose a method just for testing

Reply:
Hmm.. this codepath is already getting tested out through other tests here. I can probably remove these tests